### ISSUE TITLE
Adj Close - Not Factoring in Dividends

### ISSUE THREAD

**Issue by chuffly** [ranaroussi/fix-yahoo-finance#6]
Your work here is much appreciated.  I don't know if you've noticed, but Yahoo! appears to have messed with their Adj Close such that it's now the exact same as the Close column for tickers that haven't experienced a stock split in a given time period.  In other words, they aren't actually making the adjustments factoring in dividends anymore.

So this isn't an issue per se, but a request - if you could add in the functionality to either pull the dividend data into separate DataFrames via a new function or just pull it and automatically calculate it into the Adj Close column of the DataFrame output by your get_data_yahoo() function that would be awesome.  Latter request is more work than the former, but just easily pulling the Dividend data into a DataFrame so I can calculate on my own would be great.

I have been working on hacking this into a copy of your script on my desktop to see if I can't build this functionality on my own, but I'm still very much an amateur so it's slow going.

**Comment by Hasgassa** [comment]
Ran Aroussi , thank you for your work, it is helping me a lot.
Regarding to chuffly request, is it possible to provide yahoo-actions (Dividends and Splits) as they were being provided before?
Maybe it is already implemented and I don't know how to retrieve it?
Thank you!

**Comment by ranaroussi** [comment]
Working on it :)

**Comment by ranaroussi** [comment]
During my tests I've downloaded AAPL daily data (including splits and dividends), and it looks like the Adj Close is factoring in splits and dividends.

In any case, I've added the option to download dividend and stock splits data by adding ``actions=True`` to the download method.

Upgrade using:

```bash
$ pip install fix_yahoo_finance --upgrade --no-cache-dir
```

LMK if this is helping.

**Comment by Hasgassa** [comment]
Thank you very much, it helps a lot. But... :)

Before needing the fix I used to get actions independently. Only the dates with real actions (dividends or splits) were returned using web.DataReader(ticker, 'yahoo-actions', startdate, enddate).
With your fix we get all dates and need to process it all to find the important ones.

Nice to have: Is it possible to create a function to get actions separately, with the important dates only? Or return only the important dates using the structure below?

The dataframe structure used to be: Date / Action / Value. Example:
Date | Action | Value
1997-05-29	DIVIDEND	0.093750
1997-07-14	SPLIT	0.500000

Anyway, the current solution is already great and takes all the info in 1 interaction! :)
Thank you very much!

**Comment by ranaroussi** [comment]
In 0.0.11, use ``actions='only'`` to get a only actions ;-)

Upgrade using:

```bash
$ pip install fix_yahoo_finance --upgrade --no-cache-dir
```

**Comment by Hasgassa** [comment]
Many thanks once again. Last question :)

For 2 years period:
data = pdr.get_data_yahoo('AAPL', actions='only', start='2015-05-01', end='2017-05-31') - **4 columns**
           **Stock Splits**    action  value
Date                                    
2015-05-07          **NaN**  DIVIDEND   0.52
2015-08-06          **NaN**  DIVIDEND   0.52
...
2017-05-11  0.630000  DIVIDEND
(_Maybe it occurs when SPLIT exists for ticker but is not displayed cause of the period chosen?_)

For longer period:
pdr.get_data_yahoo('AAPL', actions='only', start='2010-05-01', end='2017-05-31') - **3 columns**
               value    action
Date                          
2012-08-09  0.378570  DIVIDEND
2012-11-07  0.378570  DIVIDEND
...
2014-06-09  0.142857     SPLIT
...
(_Now we have a split on the query, Stock Splits columns is not show anymore as expected._)

Little bug?

**Comment by ranaroussi** [comment]
Yep. Fixed in 0.0.12 :)

**Comment by chuffly** [comment]
Excellent, much appreciated!  

Yeah, Yahoo! appears to have fixed their Adj Close calculations midway through last week.  Then it went out for a day, and then they fixed it again.  So far it's working again, but this feature will be nice to have given that Yahoo! has been pretty unreliable lately ...

Thanks for your work on this script, it's been a huge help for grabbing data for my trend following models.
